inst/odin/lancelot.R

```diff
@@ -2482,84 +2482,629 @@ update(hfr_age[]) <- sum(new_HFR_disag_weighted_inc[i, , ]) /
+     (1 - pillar2_specificity) * mod_pillar2_80_plus_neg) /
+  (mod_pillar2_80_plus_pos + mod_pillar2_80_plus_neg)
+compare(pillar2_80_plus_pos) ~
+  beta_binomial(pillar2_80_plus_tot, mod_pillar2_80_plus_prob_pos,
+                rho_pillar2_tests)
+
+pillar2_over25_pos <- data()
+pillar2_over25_tot <- data()
+mod_pillar2_over25_pos <- sympt_cases_over25_inc + rexp(exp_noise)
+mod_pillar2_over25_neg <-
+  p_NC_today_25_49 * (N_tot_25_49 - sympt_cases_25_49_inc) +
+  p_NC_today_50_64 * (N_tot_50_64 - sympt_cases_50_64_inc) +
+  p_NC_today_65_79 * (N_tot_65_79 - sympt_cases_65_79_inc) +
+  p_NC_today_80_plus * (N_tot_80_plus - sympt_cases_80_plus_inc) +
+  rexp(exp_noise)
+mod_pillar2_over25_prob_pos <-
+  (pillar2_sensitivity * mod_pillar2_over25_pos +
+     (1 - pillar2_specificity) * mod_pillar2_over25_neg) /
+  (mod_pillar2_over25_pos + mod_pillar2_over25_neg)
+compare(pillar2_over25_pos) ~
+  beta_binomial(pillar2_over25_tot, mod_pillar2_over25_prob_pos,
+                rho_pillar2_tests)
+
+pillar2_pos <- data()
+pillar2_tot <- data()
+mod_pillar2_pos <- sympt_cases_inc + rexp(exp_noise)
+mod_pillar2_neg <-
+  p_NC_today_under15 * (N_tot_under15 - sympt_cases_under15_inc) +
+  p_NC_today_15_24 * (N_tot_15_24 - sympt_cases_15_24_inc) +
+  p_NC_today_25_49 * (N_tot_25_49 - sympt_cases_25_49_inc) +
+  p_NC_today_50_64 * (N_tot_50_64 - sympt_cases_50_64_inc) +
+  p_NC_today_65_79 * (N_tot_65_79 - sympt_cases_65_79_inc) +
+  p_NC_today_80_plus * (N_tot_80_plus - sympt_cases_80_plus_inc) +
+  rexp(exp_noise)
+mod_pillar2_prob_pos <-
+  (pillar2_sensitivity * mod_pillar2_pos +
+     (1 - pillar2_specificity) * mod_pillar2_neg) /
+  (mod_pillar2_pos + mod_pillar2_neg)
+compare(pillar2_pos) ~
+  beta_binomial(pillar2_tot, mod_pillar2_prob_pos, rho_pillar2_tests)
+
+## Pillar 2 cases
+kappa_pillar2_cases <- user()
+
+pillar2_under15_cases <- data()
+phi_pillar2_cases_under15 <- user()
+phi_pillar2_cases_weekend_under15 <- user()
+phi_pillar2_cases_today_under15 <-
+  if ((time + 3) %% 7 < 2) phi_pillar2_cases_weekend_under15 else
+    phi_pillar2_cases_under15
+mod_pillar2_cases_under15 <-
+  phi_pillar2_cases_today_under15 * sympt_cases_under15_inc + rexp(exp_noise)
+compare(pillar2_under15_cases) ~
+  negative_binomial_mu(kappa_pillar2_cases, mod_pillar2_cases_under15)
+
+pillar2_15_24_cases <- data()
+phi_pillar2_cases_15_24 <- user()
+phi_pillar2_cases_weekend_15_24 <- user()
+phi_pillar2_cases_today_15_24 <-
+  if ((time + 3) %% 7 < 2) phi_pillar2_cases_weekend_15_24 else
+    phi_pillar2_cases_15_24
+mod_pillar2_cases_15_24 <-
+  phi_pillar2_cases_today_15_24 * sympt_cases_15_24_inc + rexp(exp_noise)
+compare(pillar2_15_24_cases) ~
+  negative_binomial_mu(kappa_pillar2_cases, mod_pillar2_cases_15_24)
+
+pillar2_25_49_cases <- data()
+phi_pillar2_cases_25_49 <- user()
+phi_pillar2_cases_weekend_25_49 <- user()
+phi_pillar2_cases_today_25_49 <-
+  if ((time + 3) %% 7 < 2) phi_pillar2_cases_weekend_25_49 else
```

Comment:
This could be made clearer by writing:

```r
is_weekend <- (time + 3) %% 7 < 2
```

then using

```
  if (is_weekend) phi_pillar2_cases_weekend_25_49 else ...
```